content/source/docs/plugin/framework/plan-modification.html.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: "extend"
+page_title: "Plugin Development - Framework: Plan Modification"
+description: |-
+  How to modify plan values and behaviors using the provider development framework.
+---
+
+# Plan Modification
+
+After [validation](./validation.html), but prior to applying configuration changes, Terraform generates a plan which describes the expected values and behaviors of those changes. An accurate plan helps practitioners decide on whether the configuration changes match against intended expectations. Providers are given an opportunity to tailor the plan to match the expected end state of applying the changes, such as filling in unknown values with expected known values or marking that a resource must be replaced to perform the changes. The framework refers to this support as plan modification, which can be performed against an attribute or for an entire resource.
+
+## Concepts and Implicit Behaviors
+
+Terraform and the framework support two types of plan modification on resources:
+
+- Adjusting attribute values, such as filling in a known remote default value when a configuration is not present.
+- Marking a resource as needing replacement, such as when an in-place update is not supported for a change.
+
+When the provider receives a request to generate the plan for a resource change via the framework, the following occurs:
+
+1. If the plan differs from the current resource state, any computed attributes that are null in the configuration are marked as unknown in the plan. This is intended to prevent unexpected Terraform errors, however as an enhancement providers can later fill in any values that may be known.
```

Comment:
```suggestion
1. If the plan differs from the current resource state, Terraform marks computed attributes that are null in the configuration as unknown in the plan. This is intended to prevent unexpected Terraform errors. Providers can later enter any values that may be known.
```

Reply:
Ah! Good suggestions here, just want to be careful about saying "Terraform" doing it here as its the framework, not the CLI, that is doing it.

```suggestion
1. If the plan differs from the current resource state, the framework marks computed attributes that are null in the configuration as unknown in the plan. This is intended to prevent unexpected Terraform errors. Providers can later enter any values that may be known.
```